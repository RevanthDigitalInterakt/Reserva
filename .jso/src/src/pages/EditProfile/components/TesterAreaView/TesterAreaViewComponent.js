  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.default = undefined;
  var _asyncToGenerator2 = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[1]));
  var _slicedToArray2 = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[2]));
  var _reactNative = _$$_REQUIRE(_dependencyMap[3]);
  var _react = _interopRequireWildcard(_$$_REQUIRE(_dependencyMap[4]));
  var _useAsyncStorageProvider = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[5]));
  var _EventProvider = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[6]));
  function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
  function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
  function TesterAreaViewComponent(_ref) {
    var handleToggleModalTesting = _ref.handleToggleModalTesting;
    var _useState = (0, _react.useState)(''),
      _useState2 = (0, _slicedToArray2.default)(_useState, 2),
      oneSignalToken = _useState2[0],
      setOneSignalToken = _useState2[1];
    var _useState3 = (0, _react.useState)(false),
      _useState4 = (0, _slicedToArray2.default)(_useState3, 2),
      isTesting = _useState4[0],
      setIsTesting = _useState4[1];
    var _useAsyncStorageProvi = (0, _useAsyncStorageProvider.default)(),
      getItem = _useAsyncStorageProvi.getItem,
      setItem = _useAsyncStorageProvi.setItem;
    var _useBagStore = (0, _$$_REQUIRE(_dependencyMap[7]).useBagStore)(['orderFormId']),
      orderFormId = _useBagStore.orderFormId;
    var handleChangeTesting = (0, _react.useCallback)(/*#__PURE__*/function () {
      var _ref2 = (0, _asyncToGenerator2.default)(function* (currentValue) {
        handleToggleModalTesting();
        yield setItem('isTesting', currentValue);
        setIsTesting(currentValue);
      });
      return function (_x) {
        return _ref2.apply(this, arguments);
      };
    }(), []);
    var getTokenOneSignal = (0, _react.useCallback)(/*#__PURE__*/(0, _asyncToGenerator2.default)(function* () {
      var responseToken = yield _EventProvider.default.OneSignal.getDeviceState();
      setOneSignalToken((responseToken == null ? undefined : responseToken.userId) || '');
    }), []);
    var handleInitStateToggles = (0, _react.useCallback)(/*#__PURE__*/(0, _asyncToGenerator2.default)(function* () {
      var isTesterUser = (yield getItem('isTesting')) || false;
      setIsTesting(isTesterUser);
    }), []);
    (0, _$$_REQUIRE(_dependencyMap[8]).useFocusEffect)((0, _react.useCallback)(function () {
      (0, _asyncToGenerator2.default)(function* () {
        yield getTokenOneSignal();
        yield handleInitStateToggles();
      })();
    }, []));
    return /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[9]).jsxs)(_$$_REQUIRE(_dependencyMap[10]).Box, {
      mb: "sm",
      mt: "sm",
      children: [/*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[9]).jsx)(_$$_REQUIRE(_dependencyMap[10]).Box, {
        mb: "nano",
        mt: "nano",
        children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[9]).jsx)(_reactNative.TouchableOpacity, {
          testID: "com.usereserva:id/testerAreaView_button_copy_onesignal_token",
          onPress: function onPress() {
            return (0, _$$_REQUIRE(_dependencyMap[11]).handleCopyTextToClipboard)(oneSignalToken);
          },
          children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[9]).jsx)(_$$_REQUIRE(_dependencyMap[12]).Typography, {
            testID: "com.usereserva:id/testerAreaView_onesignal_token",
            children: oneSignalToken
          })
        })
      }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[9]).jsx)(_$$_REQUIRE(_dependencyMap[10]).Box, {
        mb: "nano",
        mt: "nano",
        children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[9]).jsx)(_reactNative.TouchableOpacity, {
          testID: "com.usereserva:id/testerAreaView_button_copy_orderform_id",
          onPress: function onPress() {
            return (0, _$$_REQUIRE(_dependencyMap[11]).handleCopyTextToClipboard)(orderFormId || '');
          },
          children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[9]).jsx)(_$$_REQUIRE(_dependencyMap[12]).Typography, {
            testID: "com.usereserva:id/testerAreaView_orderform_id",
            children: orderFormId || ''
          })
        })
      }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[9]).jsxs)(_$$_REQUIRE(_dependencyMap[10]).Box, {
        flexDirection: "row",
        marginY: "micro",
        alignItems: "center",
        children: [/*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[9]).jsx)(_$$_REQUIRE(_dependencyMap[10]).Box, {
          flex: 1,
          children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[9]).jsx)(_$$_REQUIRE(_dependencyMap[12]).Typography, {
            variant: "subtituloSessoes",
            children: "Ambiente de testes"
          })
        }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[9]).jsx)(_$$_REQUIRE(_dependencyMap[10]).Box, {
          marginLeft: "micro",
          children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[9]).jsx)(_$$_REQUIRE(_dependencyMap[13]).Toggle, {
            testID: "com.usereserva:id/testerAreaView_button_toogle_tester_user",
            onValueChange: handleChangeTesting,
            thumbColor: "vermelhoAlerta",
            color: "preto",
            value: isTesting
          })
        })]
      })]
    });
  }
  var _default = exports.default = TesterAreaViewComponent;
