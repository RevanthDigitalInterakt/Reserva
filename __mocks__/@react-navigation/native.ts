const navigate = jest.fn();

const useNavigation = () => ({
  navigate,
});

export {
  useNavigation,
};
