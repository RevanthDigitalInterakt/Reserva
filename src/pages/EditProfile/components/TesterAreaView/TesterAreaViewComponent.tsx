import { Box, Toggle, Typography } from '@usereservaapp/reserva-ui';
import { TouchableOpacity } from 'react-native';
import React, { useCallback, useEffect, useState } from 'react';
import handleCopyTextToClipboard from '../../../../utils/CopyToClipboard';
import { useCart } from '../../../../context/CartContext';
import useAsyncStorageProvider from '../../../../hooks/useAsyncStorageProvider';
import EventProvider from '../../../../utils/EventProvider';

interface ITesterAreaViewComponentProps {
  handleToggleModalTesting: () => void
}

function TesterAreaViewComponent({
  handleToggleModalTesting,
}: ITesterAreaViewComponentProps): JSX.Element {
  const [oneSignalToken, setOneSignalToken] = useState<string>('');
  const [isTesting, setIsTesting] = useState<boolean>(false);
  const [onboardingView, setOnboardingView] = useState<boolean>(false);

  const { getItem, setItem } = useAsyncStorageProvider();

  const { orderForm } = useCart();

  const handleChangeTesting = useCallback(async (currentValue: boolean) => {
    handleToggleModalTesting();
    await setItem('isTesting', currentValue);
    setIsTesting(currentValue);
  }, []);

  const handleViewOnboarding = useCallback(async (currentValue: boolean) => {
    await setItem('isAppFirstLaunched', currentValue);
    setOnboardingView(currentValue);
  }, []);

  const getTokenOneSignal = useCallback(async () => {
    const responseToken = await EventProvider.OneSignal.getDeviceState();
    setOneSignalToken(responseToken?.userId || '');
  }, []);

  const handleInitStateToggles = useCallback(async () => {
    const isTesterUser = await getItem('isTesting') || false;
    setIsTesting(isTesterUser);

    const onboarding = await getItem('isAppFirstLaunched') || false;
    setOnboardingView(onboarding);
  }, []);

  useEffect(() => {
    (async () => {
      await getTokenOneSignal();
      await handleInitStateToggles();
    })();
  }, []);

  return (
    <Box mb="sm" mt="sm">
      <Box mb="nano" mt="nano">
        <TouchableOpacity testID="testerAreaView_button_copy_onesignal_token" onPress={() => handleCopyTextToClipboard(oneSignalToken)}>
          <Typography testID="testerAreaView_onesignal_token">{oneSignalToken}</Typography>
        </TouchableOpacity>
      </Box>
      <Box mb="nano" mt="nano">
        <TouchableOpacity testID="testerAreaView_button_copy_orderform_id" onPress={() => handleCopyTextToClipboard(orderForm?.orderFormId || '')}>
          <Typography testID="testerAreaView_orderform_id">{orderForm?.orderFormId || ''}</Typography>
        </TouchableOpacity>
      </Box>
      <Box flexDirection="row" marginY="micro" alignItems="center">
        <Box flex={1}>
          <Typography variant="subtituloSessoes">
            Ambiente de testes
          </Typography>
        </Box>
        <Box marginLeft="micro">
          <Toggle
            testID="testerAreaView_button_toogle_tester_user"
            onValueChange={handleChangeTesting}
            thumbColor="vermelhoAlerta"
            color="preto"
            value={isTesting}
          />
        </Box>
      </Box>
      <Box flexDirection="row" marginY="micro" alignItems="center">
        <Box flex={1}>
          <Typography variant="subtituloSessoes">
            Ativar Onboarding
          </Typography>
        </Box>
        <Box marginLeft="xxxs">
          <Toggle
            testID="testerAreaView_button_toogle_onboarding_view"
            onValueChange={handleViewOnboarding}
            thumbColor="vermelhoAlerta"
            color="preto"
            value={onboardingView}
          />
        </Box>
      </Box>
    </Box>
  );
}

export default TesterAreaViewComponent;
