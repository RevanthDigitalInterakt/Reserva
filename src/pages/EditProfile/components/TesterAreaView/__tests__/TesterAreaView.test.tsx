import { ThemeProvider } from 'styled-components/native';
import React from 'react';
import
{
  act, fireEvent, render, screen,
} from '@testing-library/react-native';
import Clipboard from '@react-native-clipboard/clipboard';
import AsyncStorage from '@react-native-async-storage/async-storage';
import TesterAreaViewComponent from '../TesterAreaViewComponent';

const TestsIds = {
  buttons: {
    oneSignal: 'testerAreaView_button_copy_onesignal_token',
    oderFormId: 'testerAreaView_button_copy_orderform_id',
    userTester: 'testerAreaView_button_toogle_tester_user',
    onboardView: 'testerAreaView_button_toogle_onboarding_view',
  },

  texts: {
    oneSignal: 'testerAreaView_onesignal_token',
    orderFormId: 'testerAreaView_orderform_id',
  },
} as const;

const oneSignalToken = '1234' as const;
const orderFormId = '123456' as const;

jest.mock('react-native-onesignal', () => ({
  default: jest.fn(),
  getDeviceState: () => Promise.resolve({
    userId: oneSignalToken,
  }),
}));

const handleToggleModalTesting = jest.fn();

const TestingComponent = (
  <ThemeProvider theme={theme}>
    <TesterAreaViewComponent handleToggleModalTesting={handleToggleModalTesting} />
  </ThemeProvider>
);

// TODO check test broken
describe.skip('ChangeFileModal', () => {
  beforeEach(async () => {
    render(TestingComponent);
  });

  it('renders without error and match snapshot', async () => {
    expect(screen.toJSON()).toMatchSnapshot();
  });

  // describe('OneSignal Token tests', () => {
  it('check pressable handle copy one signal token', async () => {
    await act(async () => {
      await fireEvent.press(screen.getByTestId(TestsIds.buttons.oneSignal));
    });

    expect(Clipboard.setString).toHaveBeenCalledTimes(1);
  });

  it('check oneSignal token when pressable handle copy oneSignal token', async () => {
    await act(async () => {
      await fireEvent.press(screen.getByTestId(TestsIds.buttons.oneSignal));
    });

    expect(Clipboard.setString).toHaveBeenCalledWith(oneSignalToken);
  });

  it('check render oneSignal token text', async () => {
    expect(screen.getByTestId(TestsIds.texts.oneSignal).children[0]).toBe(oneSignalToken);
  });
  // });

  // describe('OrderFormId tests', () => {
  it.skip('check pressable handle copy orderFormId', async () => {
    await act(async () => {
      await fireEvent.press(screen.getByTestId(TestsIds.buttons.oderFormId));
    });

    expect(Clipboard.setString).toHaveBeenCalledTimes(1);
  });

  it('check orderFormId when pressable handle copy orderFormId', async () => {
    await act(async () => {
      await fireEvent.press(screen.getByTestId(TestsIds.buttons.oderFormId));
    });

    expect(Clipboard.setString).toHaveBeenCalledWith(orderFormId);
  });

  it('check render orderFormId text', async () => {
    expect(screen.getByTestId(TestsIds.texts.orderFormId).children[0]).toBe(orderFormId);
  });
  // });

  // describe('Teste user toogle', () => {
  it('check pressable handle toogle user test', async () => {
    await act(async () => {
      await fireEvent.press(screen.getByTestId(TestsIds.buttons.userTester));
    });

    expect(AsyncStorage.setItem).toHaveBeenCalledTimes(1);
  });
  // });

  // describe('Test onboardingView toogle', () => {
  it('check pressable handle toogle user test', async () => {
    await act(async () => {
      await fireEvent.press(screen.getByTestId(TestsIds.buttons.onboardView));
    });

    expect(AsyncStorage.setItem).toHaveBeenCalledTimes(1);
  });
  // });
});
