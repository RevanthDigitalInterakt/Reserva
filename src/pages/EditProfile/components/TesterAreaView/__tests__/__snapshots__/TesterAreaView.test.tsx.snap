// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangeFileModal renders without error and match snapshot 1`] = `
<View
  mb="sm"
  mt="sm"
  style={
    Array [
      Object {
        "marginBottom": 40,
        "marginTop": 40,
      },
    ]
  }
>
  <View
    mb="nano"
    mt="nano"
    style={
      Array [
        Object {
          "marginBottom": 8,
          "marginTop": 8,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="com.usereserva:id/testerAreaView_button_copy_onesignal_token"
    >
      <Text
        style={
          Array [
            Object {},
          ]
        }
        testID="com.usereserva:id/testerAreaView_onesignal_token"
      >
        1234
      </Text>
    </View>
  </View>
  <View
    mb="nano"
    mt="nano"
    style={
      Array [
        Object {
          "marginBottom": 8,
          "marginTop": 8,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="com.usereserva:id/testerAreaView_button_copy_orderform_id"
    >
      <Text
        style={
          Array [
            Object {},
          ]
        }
        testID="com.usereserva:id/testerAreaView_orderform_id"
      >
        
      </Text>
    </View>
  </View>
  <View
    alignItems="center"
    flexDirection="row"
    marginY="micro"
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 12,
          "marginTop": 12,
        },
      ]
    }
  >
    <View
      flex={1}
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "ReservaSerif-Regular",
              "fontSize": 20,
            },
          ]
        }
        variant="subtituloSessoes"
      >
        Ambiente de testes
      </Text>
    </View>
    <View
      marginLeft="micro"
      style={
        Array [
          Object {
            "marginLeft": 12,
          },
        ]
      }
    >
      <View
        alignItems="center"
        flexDirection="row"
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
          ]
        }
      >
        <RCTSwitch
          accessibilityRole="switch"
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onTintColor="#333333"
          style={
            Object {
              "height": 31,
              "width": 51,
            }
          }
          thumbTintColor="#eee"
          tintColor="#ccc"
          value={false}
        />
        <View
          accessible={true}
          focusable={true}
          ml={5}
          mt={2}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "marginLeft": 5,
                "marginTop": 2,
              },
            ]
          }
          testID="com.usereserva:id/testerAreaView_button_toogle_tester_user"
        >
          <Text
            style={
              Array [
                Object {},
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>
  <View
    alignItems="center"
    flexDirection="row"
    marginY="micro"
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 12,
          "marginTop": 12,
        },
      ]
    }
  >
    <View
      flex={1}
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "ReservaSerif-Regular",
              "fontSize": 20,
            },
          ]
        }
        variant="subtituloSessoes"
      >
        Ativar Onboarding
      </Text>
    </View>
    <View
      marginLeft="xxxs"
      style={
        Array [
          Object {
            "marginLeft": 16,
          },
        ]
      }
    >
      <View
        alignItems="center"
        flexDirection="row"
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
          ]
        }
      >
        <RCTSwitch
          accessibilityRole="switch"
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onTintColor="#333333"
          style={
            Object {
              "height": 31,
              "width": 51,
            }
          }
          thumbTintColor="#eee"
          tintColor="#ccc"
          value={false}
        />
        <View
          accessible={true}
          focusable={true}
          ml={5}
          mt={2}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "marginLeft": 5,
                "marginTop": 2,
              },
            ]
          }
          testID="com.usereserva:id/testerAreaView_button_toogle_onboarding_view"
        >
          <Text
            style={
              Array [
                Object {},
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
